graphql_compiler/tests/test_end_to_end.py

```diff
@@ -231,7 +232,7 @@ def test_argument_types(self) -> None:
                     validate_argument_type(arbitrary_argument_name, graphql_type, invalid_value)
 
     def test_non_null_types_pass_validation(self) -> None:
-        type_and_value = [
+        type_and_value: List[Tuple[GraphQLQueryArgumentType, Any]] = [
```

Comment:
Should that `Any` be `GraphQLQueryArgumentType` instead?

Reply:
No it should not. I verified that you get a linting error if you try to do that. 

`Any` refers to the type of value, e.g `"abc"`, `["a", "b", "c"]`. The values in this test case are all either strings or list of strings, but I decided to use the `Any` annotation here because it matches the `value` type annotation of `validate_argument_type`.

Reply:
Ah I see, I misunderstood what `GraphQLQueryArgumentType` is meant to be — how about renaming to `QueryArgumentGraphQLType` instead? `GraphQLQueryArgumentType` sounds like the "type which GraphQL query arguments are", and not the "GraphQL type of query arguments".

Also, thoughts on making a type alias for where this `Any` sits? It may or may not be a good idea, just curious what you think.

Reply:
Sounds great. 

Regarding your second point, I think that it would be a good idea to have a `QueryArgumentType` to indicate what are the allowed python types for queries. I think that it would make it clearer what the allowed python types are. (It isn't obvious that we use `datetime` for `GraphQLDatetime` arguments instead of isoformatted strings).

However, since we mainly use `Any` as the `QueryArgumentType`. I got a bunch of mypy errors when I tried to change it something more restrictive.

So I think that for this PR, I can write a basic type alias like `QueryArgumentType = Any` which I don't think has that much value or I can make the several times longer by writing `QueryArgumentType = Union[bool, ...]`.  

So I think that it's best if I open an issue for this.